backends-common/postgres/src/test/java/org/apache/james/backends/postgres/quota/PostgresQuotaCurrentValueDAOTest.java

```diff
@@ -0,0 +1,148 @@
+import org.apache.james.core.quota.QuotaType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class PostgresQuotaCurrentValueDAOTest {
+    @RegisterExtension
+    static PostgresExtension postgresExtension = PostgresExtension.withoutRowLevelSecurity(PostgresQuotaModule.MODULE);
+
+    private static final QuotaKey QUOTA_KEY = QuotaKey.of(QuotaComponent.MAILBOX, "james@abc.com", QuotaType.SIZE);
+
+    private PostgresQuotaCurrentValueDAO postgresQuotaCurrentValueDAO;
+
+    @BeforeEach
+    void setup() {
+        postgresQuotaCurrentValueDAO = new PostgresQuotaCurrentValueDAO(postgresExtension.getPostgresExecutor());
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldCreateNewRowSuccessfully() {
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, 100L).block();
+
+        assertThat(postgresQuotaCurrentValueDAO.getQuotaCurrentValue(QUOTA_KEY).block().getCurrentValue())
+            .isEqualTo(100L);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldCreateNewRowSuccessfullyWhenIncreaseAmountIsZero() {
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, 0L).block();
+
+        assertThat(postgresQuotaCurrentValueDAO.getQuotaCurrentValue(QUOTA_KEY).block().getCurrentValue())
+            .isZero();
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldIncreaseValueSuccessfully() {
+        assertThat(postgresQuotaCurrentValueDAO.getQuotaCurrentValue(QUOTA_KEY).block()).isNull();
+
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, 100L).block();
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, 100L).block();
+
+        assertThat(postgresQuotaCurrentValueDAO.getQuotaCurrentValue(QUOTA_KEY).block().getCurrentValue())
+            .isEqualTo(200L);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldDecreaseValueSuccessfullyWhenIncreaseAmountIsNegative() {
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, 200L).block();
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, -100L).block();
+
+        assertThat(postgresQuotaCurrentValueDAO.getQuotaCurrentValue(QUOTA_KEY).block().getCurrentValue())
+            .isEqualTo(100L);
+    }
+
+    @Test
+    void decreaseQuotaCurrentValueShouldDecreaseValueSuccessfully() {
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, 200L).block();
+        postgresQuotaCurrentValueDAO.decrease(QUOTA_KEY, 100L).block();
+
+        assertThat(postgresQuotaCurrentValueDAO.getQuotaCurrentValue(QUOTA_KEY).block().getCurrentValue())
+            .isEqualTo(100L);
+    }
+
+    @Test
+    void decreaseQuotaCurrentValueDownToNegativeShouldSetValueToZero() {
+        postgresQuotaCurrentValueDAO.increase(QUOTA_KEY, 100L).block();
+        postgresQuotaCurrentValueDAO.decrease(QUOTA_KEY, 1000L).block();
+
+        assertThat(postgresQuotaCurrentValueDAO.getQuotaCurrentValue(QUOTA_KEY).block().getCurrentValue())
+            .isZero();
+    }
```

Comment:
Cassandra implementation allows negative value.

Here Postgres implementation I enforce the value to never be negative. 
It is good to enforce that, no? TBH I am not sure if we accept negative quota counter and how we handle them. I feel pretty strange having a negative counter (maybe by a decreasing mistake), it is like a cheat to the quota itself (having a quota bigger than the quota).

Reply:
Please accept negative value.

Why? Because increase / decrease are likely not to be indempotant, and likely not to be retriable.

When this happens, emit a log and sanitize to 0.

We have tooling for recomputing quota.

I am unsure I want to have extremely strong consistency (transactions) only for this and likely can live with this.